packages/frontend-2/components/invite/dialog/Server.vue

```diff
@@ -0,0 +1,175 @@
+              show-label
+              :name="`project-${index}`"
+            />
+          </div>
+          <div class="relative w-4">
+            <CommonTextLink
+              v-if="fields.length > 1"
+              class="top-10 absolute right-0"
+              :class="{ 'top-7': index === 0 }"
+              @click="removeInviteItem(index)"
+            >
+              <TrashIcon class="h-4 w-4 text-foreground-2" />
+            </CommonTextLink>
+          </div>
+        </div>
+        <FormButton
+          color="subtle"
+          :icon-left="PlusIcon"
+          :disabled="anyMutationsLoading"
+          @click="addInviteItem"
+        >
+          Invite another user
+        </FormButton>
+      </div>
+    </form>
+  </LayoutDialog>
+</template>
+<script setup lang="ts">
+import type { LayoutDialogButton } from '@speckle/ui-components'
+import { useMutationLoading } from '@vue/apollo-composable'
+import { useForm, useFieldArray } from 'vee-validate'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import { useMixpanel } from '~~/lib/core/composables/mp'
+import { useServerInfo } from '~~/lib/core/composables/server'
+import { useInviteUserToProject } from '~~/lib/projects/composables/projectManagement'
+import { useInviteUserToServer } from '~~/lib/server/composables/invites'
+import { PlusIcon, TrashIcon } from '@heroicons/vue/24/outline'
+import type { InviteServerForm, InviteServerItem } from '~~/lib/invites/helpers/types'
+import { emptyInviteServerItem } from '~~/lib/invites/helpers/constants'
+import { isEmail } from '~~/lib/common/helpers/validation'
+
+const isOpen = defineModel<boolean>('open', { required: true })
+
+const { handleSubmit } = useForm<InviteServerForm>({
+  initialValues: {
+    fields: [
+      {
+        ...emptyInviteServerItem
+      }
+    ]
+  }
+})
+const {
+  fields,
+  replace: replaceFields,
+  push: pushInvite,
+  remove: removeInvite
+} = useFieldArray<InviteServerItem>('fields')
+const { mutate: inviteUserToServer } = useInviteUserToServer()
+const inviteUserToProject = useInviteUserToProject()
+const anyMutationsLoading = useMutationLoading()
+const { isAdmin } = useActiveUser()
+const { isGuestMode } = useServerInfo()
+const mixpanel = useMixpanel()
+
+const allowServerRoleSelect = computed(() => isAdmin.value || isGuestMode.value)
+const dialogButtons = computed((): LayoutDialogButton[] => [
+  {
+    text: 'Cancel',
+    props: { color: 'outline' },
+    onClick: () => {
+      isOpen.value = false
+    }
+  },
+  {
+    text: 'Invite',
+    props: {
+      submit: true,
+      disabled: anyMutationsLoading.value
+    },
+    onClick: onSubmit
+  }
+])
+
+const addInviteItem = () => {
+  pushInvite({ ...emptyInviteServerItem })
+}
+
+const removeInviteItem = (index: number) => {
+  removeInvite(index)
+}
+
+const onSubmit = handleSubmit(() => {
+  const invites = fields.value.filter((invite) => invite.value.email)
+
+  invites.forEach(async (invite) => {
```

Comment:
Can you use async with a forEach? I think there is an issue with promises here

Reply:
I'm not awaiting anything here, they can all sent out an invite async, the modal closes in any case. This is because they can all have different values (role, project etc) so they need to be seperate invites